proposals/NNNN-vehicle-type-filter.md

```diff
@@ -0,0 +1,435 @@
+
+### Configuring lock screen with Vehicle name or brand logo 
+
+ This would help apps to show the vehicle brand logo depending on vehicle type its connected too. In the current implementation, the lock screen configuration is set along with the SDL lifecycle/manager configuration. Hence while configuring the lock screen, the app does not know to which vehicle its connecting too. The apps cannot change the configuration of the lock screen once the app receives the `Register app interface` response with vehicle details. Some of the apps expressed their interest to show the vehicle brand logo on the lock screen of the app depending on vehicle type information, which is not possible with the current implementation of SDL due to reasons described above. 
+ 
+## Proposed solution
+
+This proposal will describe additional information exchange in the protocol layer and changes in Android and iOS proxy. 
+
+### Protocol Layer changes
+
+In the current implementation, every app is responsible for negotiating the maximum supported protocol version. The IVI can share vehicle type information with the proxy after protocol version negotiation. To share this info, we would need to define the new protocol message.
+
+#### Frame Info Definitions
+
+| Frame Info Value| Name | Description |
+|------------|------|-------------|
+| 0x0A| Get Vehicle Type | Requests vehicle type information|
+| 0x0B | Get Vehicle Type ACK |Acknowledges that the specific information has been shared successfully |
+| 0x0C | Get Vehicle Type NAK | Negatively acknowledges that the specific information *can not* be shared.
+
+
+1. If the system supports the `Get Vehicle type` protocol message, the proxy will send it after version negotiation.
+2. If the system does not support the `Get Vehicle type` protocol message, the proxy will send `Register App Interface` RPC after version negotiation and continue use case.
+
+#### Messaging from the app to IVI
+
+<table width="100%">
+  <tr>
+    <th>Version</th>
+    <th>E</th>
+    <th>Frame Type</th>
+    <th>Service Type</th>
+    <th>Frame Info</th>
+    <th>Session Id</th>
+    <th>Data Size</th>
+    <th>Message ID</th>
+  </tr>
+  <tr>
+    <td>5.x</td>
+    <td>no</td>
+    <td>Control</td>
+    <td>Control</td>
+    <td>Get Vehicle Type</td>
+    <td>Assigned Session</td>
+    <td>0</td>
+    <td>n</td>
+  </tr>
+  <tr>
+    <td>0b0101</td>
+    <td>0b0</td>
+    <td>0b000</td>
+    <td>0x00</td>
+    <td>0x0A</td>
+    <td>0x01</td>
+    <td>0x00000000</td>
+    <td>0x0000000n</td>
+  </tr>
+</table>
+
+#### Messaging from the IVI to the app
+
+##### Success message: IVI sends vehile type info
+
+If IVI can share vehicle type info, it will send the following protocol message.
+
+<table width="100%">
+  <tr>
+    <th>Version</th>
+    <th>E</th>
```

Comment:
Is "E" intentional here? Please update if needed.

Reply:
Yes, it is as per protocol header defined in the protocol spec. https://github.com/smartdevicelink/protocol_spec#22-version-2-frame-header